actionview/lib/action_view/template/handlers/erb.rb

```diff
@@ -72,6 +74,77 @@ def flush_newline_if_pending(src)
           end
         end
       end
+      Erubis = ActiveSupport::Deprecation::DeprecatedObjectProxy.new(erubis, "ActionView::Template::Handlers::Erubis is deprecated and will be removed from Rails 5.2.")
+
+      class Erubi < ::Erubi::Engine
+        # :nodoc: all
+        def initialize(input, properties = {})
+          @newline_pending = 0
+
+          # Dup properties so that we don't modify argument
+          properties = Hash[properties]
+          properties[:preamble]   = "@output_buffer = output_buffer || ActionView::OutputBuffer.new;"
+          properties[:postamble]  = "@output_buffer.to_s"
+          properties[:bufvar]     = "@output_buffer"
+          properties[:escapefunc] = ""
+
+          super
+        end
+
+        def evaluate(action_view_erb_handler_context)
+          pr = eval("proc { #{@src} }", binding, @filename || "(erubi)")
+          action_view_erb_handler_context.instance_eval(&pr)
+        end
+
+      private
+        def add_text(text)
+          return if text.empty?
+
+          if text == "\n"
+            @newline_pending += 1
+          else
+            src << "@output_buffer.safe_append='"
+            src << "\n" * @newline_pending if @newline_pending > 0
+            src << text.gsub(/['\\]/, '\\\\\&')
+            src << "'.freeze;"
+
+            @newline_pending = 0
+          end
+        end
+
+        BLOCK_EXPR = /\s*((\s+|\))do|\{)(\s*\|[^|]*\|)?\s*\Z/
+
+        def add_expression(indicator, code)
+          flush_newline_if_pending(src)
+
+          if (indicator == "==") || @escape
+            expr = "safe_expr_"
+          end
+
+          if BLOCK_EXPR.match?(code)
+            src << "@output_buffer.#{expr}append= " << code
+          else
+            src << "@output_buffer.#{expr}append=(" << code << ");"
+          end
```

Comment:
Frozen literal string support is not turned on yet, but this is a good idea for when it is enabled.  I'll use a slightly different approach that will save a 2-3 `<<` calls